homeassistant/components/camera/webrtc.py

```diff
@@ -0,0 +1,236 @@
+
+
+class _CameraRtspToWebRTCProvider(CameraWebRTCProvider):
+    def __init__(self, fn: RtspToWebRtcProviderType) -> None:
+        """Initialize the RTSP to WebRTC provider."""
+        self._fn = fn
+
+    async def async_is_supported(self, stream_source: str) -> bool:
+        """Return if this provider is supports the Camera as source."""
+        return any(stream_source.startswith(prefix) for prefix in _RTSP_PREFIXES)
+
+    async def async_handle_web_rtc_offer(
+        self, camera: Camera, offer_sdp: str
+    ) -> str | None:
+        """Handle the WebRTC offer and return an answer."""
+        if not (stream_source := await camera.stream_source()):
+            return None
+
+        return await self._fn(stream_source, offer_sdp, camera.entity_id)
+
+
+def async_register_rtsp_to_web_rtc_provider(
+    hass: HomeAssistant,
+    domain: str,
+    provider: RtspToWebRtcProviderType,
+) -> Callable[[], None]:
+    """Register an RTSP to WebRTC provider.
+
+    The first provider to satisfy the offer will be used.
+    """
+    provider_instance = _CameraRtspToWebRTCProvider(provider)
+    return async_register_webrtc_provider(hass, provider_instance)
+
+
+async def _async_refresh_providers(hass: HomeAssistant) -> None:
+    """Check all cameras for any state changes for registered providers."""
+
+    component: EntityComponent[Camera] = hass.data[DOMAIN]
+    await asyncio.gather(
+        *(camera.async_refresh_providers() for camera in component.entities)
+    )
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "camera/webrtc/get_client_config",
+        vol.Required("entity_id"): cv.entity_id,
+    }
+)
+@websocket_api.async_response
+async def ws_get_client_config(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """Handle get WebRTC client config websocket command."""
+    entity_id = msg["entity_id"]
+    camera = get_camera_from_entity_id(hass, entity_id)
+    if camera.frontend_stream_type != StreamType.WEB_RTC:
+        connection.send_error(
+            msg["id"],
+            "web_rtc_offer_failed",
+            (
+                "Camera does not support WebRTC,"
+                f" frontend_stream_type={camera.frontend_stream_type}"
+            ),
+        )
+        return
+
+    config = (await camera.async_get_webrtc_client_configuration()).to_dict()
+    connection.send_result(
+        msg["id"],
+        config,
+    )
+
+
+async def async_get_supported_providers(
+    hass: HomeAssistant, camera: Camera
+) -> list[CameraWebRTCProvider]:
+    """Return a list of supported providers for the camera."""
+    providers = hass.data.get(DATA_WEBRTC_PROVIDERS)
+    if not providers or not (stream_source := await camera.stream_source()):
+        return []
+
+    return [
+        provider
+        for provider in providers
+        if await provider.async_is_supported(stream_source)
+    ]
+
+
+@callback
+def register_ice_server(
```

Comment:
Shouldn't it be `register_ice_servers` and use a list of RTCIceServer?

Reply:
Currently, there is no use case to return multiple servers. One server can have multiple URLs